### PR TITLE
suit: generate manifests directly from source

### PR DESCRIPTION
**Pull request by e-rk**
Previously, the SUIT manifest templates were copied from the source
pointed by  SUIT_ENVELOPE_ROOT_TEMPLATE into the sample directory to
allow the user to modify them. The manifests were then generated from
the templates within the sample directory.

This behavior is now replaced with generating the manifests from
templates directly from the source path. There is no intermediate step
of putting the templates into the sample directory.

### CODE REVIEW

**Review by tomchy — APPROVED** [review]

File: sysbuild/Kconfig.suit [hunk 1]
```diff
@@ -35,7 +35,7 @@ config SUIT_ENVELOPE_ROOT_TEMPLATE
 
 config SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION
 	string "Path to the folder with envelope templates"
-	default "../"
+	default "."
 	help
 	  Path to the folder containing editable templates used to create binary envelopes.
 	  Input templates are created by the build system during first build from the SUIT_ENVELOPE_DEFAULT_TEMPLATE.
```
Comment (nordicjm) [inline sysbuild/Kconfig.suit:41]: Comment is wrong and should say `PROJECT_BINARY_DIR` not source dir

File: sysbuild/Kconfig.suit [hunk 2]
```diff
@@ -35,7 +35,7 @@ config SUIT_ENVELOPE_ROOT_TEMPLATE
 
 config SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION
 	string "Path to the folder with envelope templates"
-	default "../"
+	default "."
```
Comment (tejlmand) [inline sysbuild/Kconfig.suit:38]: I would much prefer this setting is complete removed.

Please fix the location where the file is created to the build folder, and not allow users to modify it.

**Review by tejlmand — CHANGES_REQUESTED** [review]

**Review by ahasztag — APPROVED** [review]

File: samples/matter/lock/prj.conf [hunk 3]
```diff
@@ -42,3 +42,6 @@ CONFIG_CHIP_APP_LOG_LEVEL=3
 # Enable Factory Data feature
 CONFIG_CHIP_FACTORY_DATA=y
 CONFIG_CHIP_FACTORY_DATA_BUILD=y
+
+# Set SUIT template paths
+CONFIG_SUIT_ENVELOPE_TEMPLATE="suit_templates/app_envelope.yaml.jinja2"
```
Comment (nordicjm) [inline samples/matter/lock/prj.conf:47]: surely this should be in an nrf54h20 overlay file?

File: samples/matter/lock/prj_release.conf [hunk 4]
```diff
@@ -53,3 +53,6 @@ CONFIG_CHIP_FACTORY_DATA_BUILD=y
 
 # Enable Watchdog
 CONFIG_NCS_SAMPLE_MATTER_WATCHDOG=y
+
+# Set SUIT template paths
+CONFIG_SUIT_ENVELOPE_TEMPLATE="suit_templates/app_envelope.yaml.jinja2"
```
Comment (nordicjm) [inline samples/matter/lock/prj_release.conf:58]: as above

File: samples/matter/lock/prj_thread_wifi_switched.conf [hunk 5]
```diff
@@ -56,3 +56,6 @@ CONFIG_ISR_TABLES_LOCAL_DECLARATION=y
 # Enforce the use of mbedTLS crypto backend instead of PSA Crypto by
 # building OpenThread from sources. Wi-Fi stack does not support PSA Crypto yet.
 CONFIG_OPENTHREAD_SOURCES=y
+
+# Set SUIT template paths
+CONFIG_SUIT_ENVELOPE_TEMPLATE="suit_templates/app_envelope.yaml.jinja2"
```
Comment (nordicjm) [inline samples/matter/lock/prj_thread_wifi_switched.conf:61]: as above

File: samples/matter/lock/sysbuild/ipc_radio/prj.conf [hunk 6]
```diff
@@ -9,3 +9,6 @@ CONFIG_UART_CONSOLE=n
 CONFIG_LOG=n
 
 CONFIG_RESET_ON_FATAL_ERROR=n
+
+# Set SUIT template paths
+CONFIG_SUIT_ENVELOPE_TEMPLATE="suit_templates/rad_envelope.yaml.jinja2"
```
Comment (nordicjm) [inline samples/matter/lock/sysbuild/ipc_radio/prj.conf:14]: as above

File: samples/matter/lock/sysbuild/ipc_radio/prj_release.conf [hunk 7]
```diff
@@ -9,3 +9,6 @@ CONFIG_UART_CONSOLE=n
 CONFIG_LOG=n
 
 CONFIG_RESET_ON_FATAL_ERROR=y
+
+# Set SUIT template paths
+CONFIG_SUIT_ENVELOPE_TEMPLATE="suit_templates/rad_envelope.yaml.jinja2"
```
Comment (nordicjm) [inline samples/matter/lock/sysbuild/ipc_radio/prj_release.conf:14]: as above

File: samples/matter/template/prj.conf [hunk 8]
```diff
@@ -45,3 +45,6 @@ CONFIG_USE_SEGGER_RTT=n
 # Enable Factory Data feature
 CONFIG_CHIP_FACTORY_DATA=y
 CONFIG_CHIP_FACTORY_DATA_BUILD=y
+
+# Set SUIT template paths
+CONFIG_SUIT_ENVELOPE_TEMPLATE="suit_templates/app_envelope.yaml.jinja2"
```
Comment (nordicjm) [inline samples/matter/template/prj.conf:50]: etc.

File: samples/matter/lock/prj.conf (hunk 3 above)
Comment (e-rk) [reply samples/matter/lock/prj.conf]: Right, moved to board overlay in this and subsequent occurrences.

**Comment by e-rk** [comment]
> Why do we need to provide each path separately for a specific image? Why was the previous way worse than the proposed approach?

The previous approach made the mistake of copying files into the sample directory during the build. What you see here is an attempt to fix an immediate problem and we will be coming back to this after the release.

**Review by ArekBalysNordic — APPROVED** [review]
Approving, since there is a plan to simplify it after the release. :)

**Review by robertstypa — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by e-rk** [comment]
@tejlmand could you review?

**Review by tejlmand — APPROVED** [review]

**Pull request merged**